### ISSUE TITLE
googlecompute: using startup_script_file hangs builder

### ISSUE THREAD

**Issue by jacek-jablonski** [hashicorp/packer#9121]
#### Overview of the Issue

Using `startup_script_file` causes googlecompute builder to hang.
`googlecompute: Startup script not finished yet. Waiting...` is repeated many times until timeout error occurs.

#### Reproduction Steps

image.pkr.hcl:
```
source "googlecompute" "runner" {
  account_file = "account.json"
  image_name          = "runner-{{timestamp}}"
  image_family        = "runner"
  project_id          = "xxxx"
  source_image_family = "ubuntu-minimal-2004-lts"
  ssh_username        = "packer"
  startup_script_file = "bootstrap.sh"
  zone                = "europe-west4-c"
}

build {
  sources = [
    "source.googlecompute.runner"
  ]
}
```
bootstrap.sh:
```
#!/usr/bin/env bash

echo "test"
exit 0
```

### Packer version

Packer v1.5.5

### Operating system and Environment details

macOS 10.15.4

### Log Fragments and crash.log files

https://gist.github.com/jacek-jablonski/a26241e9a8b24ed68648dadfe7dcf047

**Comment by nywilken** [comment]
Hi @jacek-jablonski are you still running into issues here? Does setting an explicit path to bash `#!/bin/bash` change the behavior. 

There should be information in the serial logs indicating what is happening with the startup script file. That may hint at why it’s hanging.

**Comment by nywilken** [comment]
Issue confirmed not to be related to the shell path. No need to test. I will take a look to see what might be happening here. Thanks again for reporting this issue.

**Comment by keznikl** [comment]
I've had the issue when using a custom service account which didn't have `compute.instances.setMetadata`
(I see that you're using a custom service account too, due to `account_file = "account.json"`)
So packer kept waiting forever for the custom cloud init to finish.
I think this is a bug, no other provisioners wait for cloud-init to finish this way.
Also, requiring `compute.instances.setMetadata` is a pretty demanding requirement, because this permission also let's one update gain SSH access to the VM (or other VMs sharing the same service account).

**Comment by nywilken** [comment]
Hello :wave: thanks again for reporting, this should be fixed by #9504; there are some test binaries are available via the link below if you would like to test the fix. The tl;dr is that the polling logic for the startup script was executing after Packer entered the provision state which is why Packer would warn that the the Startup Script was not complete even though the script exited successfully. 

https://app.circleci.com/pipelines/github/hashicorp/packer/5859/workflows/8d0099dd-839c-42d5-ba9b-f1d559b4cf53/jobs/63069/artifacts

@keznikl re: "I think this is a bug, no other provisioners wait for cloud-init to finish this way.

Agreed. I started work on adding a new configuration option as a first to step phase out the wrapped startup script https://github.com/hashicorp/packer/pull/9505

**Comment by nywilken** [comment]
Hi there, we've introduced a new configuration option for toggling the wrapper script on and off. For backwards compatibility the wrapper script option is enabled so out of the box all startup scripts will be wrapped and behave as they do now. By setting `"wrap_startup_script": "false"` in a Packer configuration the provided startup script will be executed directly on the GCE instance with no additional Packer wait time. The latter will become the default at a later release.  For more information on `wrap_startup_script` you can refer to the documentation [here](https://github.com/hashicorp/packer/blob/d4a7dd24d997e3d90abc16124887a63146fe60fa/builder/googlecompute/config.go#L204)

There are test binaries available for [download](https://app.circleci.com/pipelines/github/hashicorp/packer/5964/workflows/f041cb88-4105-40c0-95e8-9f77a8591d56/jobs/64512/artifacts) if you would like to test the new option.
